### ISSUE TITLE
New line intersection sometimes fail

### ISSUE THREAD

**Issue by TiltedClone** [mozman/ezdxf#128]
I have seen you are moving a part of your math entities (in this case the ezdxf.math.ConstructionLine) into a 2d section. With this you have also updated your algoritm for finding the intersection between two lines. As of ezdxf version 11+ the intersection between 2 lines seem to fail sometimes. It worked perfectly fine in version 10.4 and before.

In the attached images I draw a small circle at the point two entities intersect. As you can see the intersection is ignored at (seemingly) random points.
![dxf_line_intersect](https://user-images.githubusercontent.com/40609626/76163317-4ebf0280-6145-11ea-896f-0a5967b746a7.PNG)

**Comment by mozman** [comment]
Could you provide the code please?

**Comment by TiltedClone** [comment]
I have found some additional information about the bug.
First I have a clean file not modified then I have a file I've already run from before and modified by adding a marking layer (not checked in the intersection)
(named "modified" and "clean" in the attached zip). For ezdxf 0.10.4 the intersection is found on both the modified and clean file.
Modified (0.10.4):
![ezdxf-0 10 4](https://user-images.githubusercontent.com/40609626/76165343-98641900-6156-11ea-9d6d-324a9062d991.PNG)

But in ezdxf 11+ the intersection can only be found on the clean file.
Modified (0.11.1):
![ezdxf_fail_latest](https://user-images.githubusercontent.com/40609626/76165350-a4e87180-6156-11ea-87aa-d59e701902d8.PNG)

This bug might occur because of the way I have saved the dxf-file while modifying it but it still works in 10.4 at least. I have modified it using streaming. My knowledge on dxf files and the difference between the clean and modified one is not good enough to find the problem

Example program where the bug occur
[ezdxf_bug.zip](https://github.com/mozman/ezdxf/files/4303374/ezdxf_bug.zip)
